src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/Metadata/PE/IsConstModifierTests.vb

```diff
@@ -0,0 +1,625 @@
+
+        <Fact>
+        Public Sub ReadOnlySignaturesAreNotSupported_Indexers_Parameters_Abstract()
+            Dim reference = CreateCSharpCompilation("
+public abstract class TestRef
+{
+    public abstract int this[ref readonly int p] { set; }
+}", parseOptions:=New CSharpParseOptions(CSharp.LanguageVersion.Latest)).EmitToImageReference()
+
+            Dim source =
+                <compilation>
+                    <file>
+Class Test
+    Shared Sub Main(obj As TestRef) 
+        Dim p = 0
+        Dim value = obj(p)
+    End Sub
+End Class
+    </file>
+                </compilation>
+
+            Dim compilation = CreateCompilationWithMscorlib(source, references:={reference})
+
+            AssertTheseDiagnostics(compilation, <expected>
+BC30643: Property 'TestRef.Item(p As )' is of an unsupported type.
+        Dim value = obj(p)
+                    ~~~
+                                                </expected>)
+        End Sub
+
+        <Fact>
+        Public Sub ReadOnlySignaturesAreNotSupported_Indexers_ReturnTypes_Virtual()
+            Dim reference = CreateCSharpCompilation("
+public class TestRef
+{
+    private int value = 0;
+    public virtual ref readonly int this[int p] => ref value;
+}", parseOptions:=New CSharpParseOptions(CSharp.LanguageVersion.Latest)).EmitToImageReference()
+
+            Dim source =
+                <compilation>
+                    <file>
+Class Test
+    Shared Sub Main() 
+        Dim obj = New TestRef()
+        Dim value = obj(0)
+    End Sub
+End Class
+    </file>
+                </compilation>
+
+            Dim compilation = CreateCompilationWithMscorlib(source, references:={reference})
+
+            AssertTheseDiagnostics(compilation, <expected>
+BC30643: Property 'TestRef.Item(p As )' is of an unsupported type.
+        Dim value = obj(0)
+                    ~~~
+                                                </expected>)
+        End Sub
+
+        <Fact>
+        Public Sub ReadOnlySignaturesAreNotSupported_Indexers_ReturnTypes_Abstract()
+            Dim reference = CreateCSharpCompilation("
+public abstract class TestRef
+{
+    public abstract ref readonly int this[int p] { get; }
+}", parseOptions:=New CSharpParseOptions(CSharp.LanguageVersion.Latest)).EmitToImageReference()
+
+            Dim source =
+                <compilation>
+                    <file>
+Class Test
+    Shared Sub Main(obj As TestRef) 
+        Dim value = obj(0)
+    End Sub
+End Class
+    </file>
+                </compilation>
+
+            Dim compilation = CreateCompilationWithMscorlib(source, references:={reference})
+
+            AssertTheseDiagnostics(compilation, <expected>
+BC30643: Property 'TestRef.Item(p As )' is of an unsupported type.
+        Dim value = obj(0)
+                    ~~~
+                                                </expected>)
+        End Sub
+
+        <Fact>
+        Public Sub ReadOnlySignaturesAreNotSupported_Indexers_ReturnTypes_NoModifiers()
```

Comment:
We always place modreqs on ref readonly returns even if there are no modifiers. because consumers might try to change its value.

---
In reply to: [119781279](https://github.com/dotnet/roslyn/pull/19658#discussion_r119781279) [](ancestors = 119781279)